### PR TITLE
Add default mark II

### PR DESCRIPTION
**Pull request by kyriesent**
Here we go again!

### CODE REVIEW

File: lib/schema.js [hunk 1]
```diff
@@ -324,7 +326,7 @@ function PrimitiveSchema(type) {
 
 util.inherits(PrimitiveSchema, Schema);
 
-function FieldSchema(name, type) {
+function FieldSchema(name, type, defaultValue) {
```
Comment (blake-nicholson) [inline lib/schema.js]: The type variable is a Schema object (that now has a default value), but the FieldSchema object also has a default value.  It looks to me like FieldSchema does not need a default value, but the various Schema subclasses do.

Comment (kyriesent) [inline lib/schema.js]: I attempted to remove this argument and the `defaultValue` below and I got an AvroValidationError on the unit test for defaults. It seems like the default is actually necessary here, in particular when serializing records... am I missing something?

Comment (kyriesent) [inline lib/schema.js]: @blake-nicholson can you let me know if you have a fix for this? At this point I think we should merge the request as-is as I'm not sure how to add a fix for this without breaking the tests that come in.

Comment (blake-nicholson) [inline lib/schema.js]: @kyriesent Let me know when you have a few minutes to go through this together.  I can help you update the other code.

**Comment by blake-nicholson** [comment]
@kyriesent: I updated the code to remove the duplicate locations of default value.  It is now just in Field (which was renamed from FieldSchema).  Please review my updates.  If everything looks good to you, go ahead and merge your PR.

**Comment by kyriesent** [comment]
Ah this makes much more sense. Thanks for cleaning this up @blake-nicholson

**Pull request merged**